great_expectations/core/usage_statistics/anonymizers/anonymizer.py

```diff
@@ -1,765 +1,136 @@
+            DataConnector,
         )
 
-        return anonymized_info_dict
-
-    def anonymize_data_connector_info(self, name: str, config: dict) -> dict:
-        """Anonymize DataConnector objs from the 'great_expectations.datasource.data_connector' module.
+        return obj is not None and isinstance(obj, DataConnector)
 
-        Args:
-            name (str): The name of the connector.
-            config (dict): The dictionary configuration corresponding to the connector.
-
-        Returns:
-            An anonymized dictionary payload that obfuscates user-specific details.
-        """
-        anonymized_info_dict = {
-            "anonymized_name": self.anonymize(name),
-        }
-
-        # Roundtrip through schema validation to remove any illegal fields add/or restore any missing fields.
-        from great_expectations.data_context.types.base import (
-            DataConnectorConfig,
-            dataConnectorConfigSchema,
-        )
+    @staticmethod
+    def _is_batch_info(obj: object) -> bool:
+        from great_expectations.data_asset.data_asset import DataAsset
+        from great_expectations.validator.validator import Validator
 
-        data_connector_config: DataConnectorConfig = dataConnectorConfigSchema.load(
-            config
-        )
-        data_connector_config_dict: dict = dataConnectorConfigSchema.dump(
-            data_connector_config
-        )
+        if object is None:
+            return False
 
-        self._anonymize_object_info(
-            anonymized_info_dict=anonymized_info_dict,
-            object_config=data_connector_config_dict,
+        return isinstance(obj, (Validator, DataAsset)) or (
+            isinstance(obj, tuple) and len(obj) == 2
         )
 
-        return anonymized_info_dict
-
-    def anonymize_expectation_suite_info(
-        self, expectation_suite: "ExpectationSuite"  # noqa: F821
-    ) -> dict:
-        """Anonymize ExpectationSuite objs from the 'great_expectations.core' module.
-
-        Args:
-            expectation_suite ("ExpectationSuite"): An instance of ExpectationSuite.
-
-        Returns:
-            An anonymized dictionary payload that obfuscates user-specific details.
-        """
-        anonymized_info_dict = {}
-        anonymized_expectation_counts = list()
+    @staticmethod
+    def _is_store_info(info: dict):
+        return "store_name" in info or "store_obj" in info
 
-        expectations = expectation_suite.expectations
-        expectation_types = [
-            expectation.expectation_type for expectation in expectations
+    @staticmethod
+    def can_handle(obj: object, **kwargs) -> bool:
+        return isinstance(obj, object)
+
+    def build_init_payload(self, data_context: "DataContext") -> dict:  # noqa: F821
+        """Adds information that may be available only after full data context construction, but is useful to
+        calculate only one time (for example, anonymization)."""
```

Comment:
Move logic from `usage_statistics` to `anonymizer` - most of the method is anonymizing behavior so it makes more sense here. Additionally, this lets us keep all methods private.

Reply:
Totally works - this feels a bit weird to me though from a responsibility standpoint to include in any of the Anonymizer classes. I don't have a better idea though so just thinking out loud :)

Reply:
Ah yes this is a very valid point. To be perfectly honest, the primary purpose here is to prevent having to expose functionality from the anonymizer.

By keeping the logic here, we can keep all helpers private and maintain the ideal that `anonymize` is the single public entrypoint into anonymizer behavior.

If you feel strongly about this (and I think that's totally justifiable), let me know and I can figure something out.